cmd/crdvalidator/README.md

```diff
@@ -0,0 +1,116 @@
+# crdvalidator
+
+## Summary
+A part of the core value proposition for RukPak is the safe upgrade of `CRD`s. As a result, this repository comes equipped with a "CRD Validator Webhook", `crdvalidator`, that will validate specified `CRD` upgrade events.
+
+
+### How to use
+
+The CRD Validator webhook will ensure that any CRD upgrade is safe as long as the incoming CRD has the `core.rukpak.io/enable-safe-upgrade-validation` annotation set to `enabled` or if it is owned by RukPak via an owner reference.
+
+```yaml
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.7.0
+    core.rukpak.io/safe-upgrade-validation: enabled # <--
+# 
+# ...
+#
+```
+
+If you would like to disable validation for a CRD created by RukPak, you will need to explicitly set this annotation as `disabled` when it is added to the `Bundle`.
+
+```yaml
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.7.0
+    core.rukpak.io/safe-upgrade-validation: disabled # <--
+# 
+# ...
+#
+```
+
+
+### What is protected
+There are three main cases `crdvalidator` is checking against with every specified `CRD` upgrade. These are where a new `CRD`:
+1. Removes a version that the old `CRD` had and it must ensured that nothing is stored at the removed version.
```

Comment:
nit: `Removes a stored version and ensures that removing the stored version does not result in data loss` for example?